Server/ReasnAPI/ReasnAPI/Services/EventService.cs

```diff
@@ -1,16 +1,276 @@
+            eventToUpdate.OrganizerId = eventDto.OrganizerId;
+            eventToUpdate.StartAt = eventDto.StartAt;
+            eventToUpdate.EndAt = eventDto.EndAt;
+            eventToUpdate.UpdatedAt = DateTime.Now;
+            eventToUpdate.StatusId = eventDto.StatusId;
+
+            context.Events.Update(eventToUpdate);
+            context.SaveChanges();
+
+            var existingTags = context.EventTags.Where(r => r.EventId == eventId).Include(eventTag => eventTag.Tag).ToList();
+
+            if (eventDto.Tags == null)
+            {
+                return eventDto;
+            }
+
+            var newTags = eventDto.Tags
+                .Select(tagDto => new Tag
+                {
+                    Name = tagDto.Name,
+                }).ToList();
+
+            var tagsToRemove = from existingTag in existingTags
+                               where newTags.All(r => r.Name != existingTag.Tag.Name)
+                               select existingTag;
+
+            foreach (var existingTag in tagsToRemove)
+            {
+                context.EventTags.Remove(existingTag);
+            }
+
+            var tagsToAdd = from newTag in newTags
+                            where existingTags.All(r => r.Tag.Name != newTag.Name)
+                            select newTag;
+
+            foreach (var newTag in tagsToAdd)
+            {
+                var existingTag = context.Tags.FirstOrDefault(t => t.Name == newTag.Name);
+                if (existingTag == null)
+                {
+                    existingTag = new Tag { Name = newTag.Name };
+                    context.Tags.Add(existingTag);
+                    context.SaveChanges();
+                }
+
+                context.EventTags.Add(new EventTag { EventId = eventId, TagId = existingTag.Id });
+            }
+
+            context.SaveChanges();
+        }
+
+        return eventDto;
+    }
+
+    public bool DeleteEvent(int eventId)
+    {
+        using (var scope = new TransactionScope())
+        {
+            var eventToDelete = context.Events.FirstOrDefault(r => r.Id == eventId);
+
+            if (eventToDelete == null)
+            {
+                return false;
+            }
+
+            context.EventTags.RemoveRange(context.EventTags.Where(r => r.EventId == eventId));
+            context.Events.Remove(eventToDelete);
+
+            context.SaveChanges();
+        }
+
+        return true;
     }
-}
+
+    public EventDto GetEventById(int eventId)
+    {
+        var eventToReturn = context.Events.FirstOrDefault(r => r.Id == eventId);
+        if (eventToReturn == null)
+        {
+            return null;
+        }
+
+        var tagIds = context.EventTags
+            .Where(r => r.EventId == eventId)
+            .Select(r => r.TagId)
+            .ToList();
+
+        var tags = context.Tags
+            .Where(t => tagIds.Contains(t.Id))
+            .Select(t => new TagDto { Name = t.Name })
+            .ToList();

```

Comment:
Hmm, think that there might be an easier way to do it in single query, single variable, maybe after fixing bridge table it will be more obvious.